src/main/java/codoc/logic/commands/EditCommand.java

```diff
@@ -195,6 +233,164 @@ public void setModules(Set<Module> modules) {
             this.modules = (modules != null) ? new HashSet<>(modules) : null;
         }
 
+        /**
+         * Adds {@code skills} to this object's {@code skills}.
+         * A defensive copy of {@code skills} is used internally.
+         */
+        public void addSkills(Set<Skill> skills) {
+            Set<Skill> set = new HashSet<>();
+            if (skills != null && this.skills != null) {
+                set.addAll(this.skills);
+                set.addAll(skills);
+                this.skills = set;
+            } else {
+                this.skills = skills;
+            }
+        }
+
+        /**
+         * Deletes {@code skills} from this object's {@code skills}.
+         * A defensive copy of {@code skills} is used internally.
+         */
+        public void deleteSkills(Set<Skill> skills) {
+            if (skills != null && skills.size() == 0) {
+                this.skills = skills;
+            } else if (skills != null && this.skills != null) {
+                deleteSkillsHelper(skills);
+            }
+        }
+
+        private void deleteSkillsHelper(Set<Skill> skills) {
+            if (hasSkills(skills)) {
+                Set<Skill> result = new HashSet<>();
+                for (Skill s : this.skills) {
+                    if (!skills.contains(s)) {
+                        result.add(s);
+                    }
+                }
+                this.skills = result;
+            } else {
+                this.skills = null;
+            }
+        }
+
+        /**
+         * Updates {@code skills} from this object's {@code skills}.
+         * A defensive copy of {@code skills} is used internally.
+         */
+        public void updateSkills(Set<Skill> oldSkills, Set<Skill> newSkills) {
+            if (oldSkills != null && newSkills != null && this.skills != null) {
+                updateSkillsHelper(oldSkills, newSkills);
+            }
+        }
+
+        private void updateSkillsHelper(Set<Skill> oldSkills, Set<Skill> newSkills) {
+            if (hasSkills(oldSkills)) {
+                Set<Skill> result = new HashSet<>();
+                for (Skill s : this.skills) {
+                    if (oldSkills.contains(s)) {
+                        Skill newSkill = newSkills.iterator().next();
+                        result.add(newSkill);
+                        newSkills.remove(newSkill);
+                    } else {
+                        result.add(s);
+                    }
+                }
+                this.skills = result;
+            } else {
+                this.skills = null;
+            }
+        }
```

Comment:
Perhaps can consider breaking this method down into shorter methods to avoid nesting and 'arrowhead' style code. There are a few of these in this file.